src/main/java/bio/terra/service/tabulardata/azure/StorageTableDao.java

```diff
@@ -0,0 +1,192 @@
+package bio.terra.service.tabulardata.azure;
+
+import bio.terra.model.BulkLoadFileState;
+import bio.terra.model.BulkLoadHistoryModel;
+import bio.terra.service.snapshot.exception.CorruptMetadataException;
+import bio.terra.service.tabulardata.LoadHistoryUtil;
+import com.azure.data.tables.TableClient;
+import com.azure.data.tables.TableServiceClient;
+import com.azure.data.tables.models.ListEntitiesOptions;
+import com.azure.data.tables.models.TableEntity;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+
+@Component
+public class StorageTableDao {
+
+  private static final String LOAD_HISTORY_TABLE_NAME_SUFFIX = "LoadHistory";
+
+  /**
+   * Store the results of a bulk file load in an Azure Storage Table
+   *
+   * <p>The table name will be the result of the dataset id passed through {@link
+   * StorageTableDao#toStorageTableNameFromUUID} Entities will be partitioned on the loadTag and
+   * their row keys will be the value of {@link BulkLoadHistoryModel#getFileId()}
+   *
+   * @param serviceClient A service client for the dataset
+   * @param datasetId the id of the dataset
+   * @param loadTag Load tag to partition on
+   * @param loadTime The time the load occurred
+   * @param loadHistoryArray The models to store
+   */
+  public void loadHistoryToAStorageTable(
```

Comment:
The class name is `StorageTable` so you don't need to repeat it here, a reader can assume that all methods in this class deal with storage tables. Like you have `getLoadHistory()`, this can be `storeLoadHistory()` or `putLoadHistory()`.